Running tests with no gcloud project set fails

I was running this locally, and hit the following:

```
======================================================================
FAIL: test_unsupported_schema (test_mltoolbox.test_package_functions.TestAnalyze)
Test supported schema values.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/google/home/craigcitro/gh/pydatalab/solutionbox/structured_data/test_mltoolbox/test_package_functions.py", line 113, in test_unsupported_schema
    job.fatal_error.message)
AssertionError: 'Schema contains an unsupported type bytes.' not found in 'No project ID found. Perhaps you should set one by running"%datalab project set -p <project-id>" in a code cell.'

----------------------------------------------------------------------
```

Setting a project ID (via `gcloud config set project myproject`) fixed the issue.

/cc @brandondutra

--- a/solutionbox/structured_data/mltoolbox/_structured_data/_package.py
+++ b/solutionbox/structured_data/mltoolbox/_structured_data/_package.py
@@ -189,14 +189,17 @@ def _analyze(output_dir, dataset, cloud=False, project_id=None):
                                            prefix='schema')
     file_io.write_string_to_file(schema_file_path, json.dumps(dataset.schema))
 
+    #TODO(brandondutra) use project_id in the local preprocess function.
     args = ['preprocess',
             '--input-file-pattern=%s' % dataset.input_files[0],
             '--output-dir=%s' % output_dir,
             '--schema-file=%s' % schema_file_path]
 
     if cloud:
+      if not project_id:
+        project_id = _default_project()
       print('Track BigQuery status at')
-      print('https://bigquery.cloud.google.com/queries/%s' % _default_project())
+      print('https://bigquery.cloud.google.com/queries/%s' % project_id)
       preprocess.cloud_preprocess.main(args)
     else:
       preprocess.local_preprocess.main(args)

--- a/solutionbox/structured_data/mltoolbox/_structured_data/_package.py
+++ b/solutionbox/structured_data/mltoolbox/_structured_data/_package.py
@@ -189,7 +189,7 @@ def _analyze(output_dir, dataset, cloud=False, project_id=None):
                                            prefix='schema')
     file_io.write_string_to_file(schema_file_path, json.dumps(dataset.schema))
 
-    #TODO(brandondutra) use project_id in the local preprocess function.
+    # TODO(brandondutra) use project_id in the local preprocess function.
     args = ['preprocess',
             '--input-file-pattern=%s' % dataset.input_files[0],
             '--output-dir=%s' % output_dir,

--- a/solutionbox/structured_data/test_mltoolbox/test_package_functions.py
+++ b/solutionbox/structured_data/test_mltoolbox/test_package_functions.py
@@ -108,7 +108,8 @@ class TestAnalyze(unittest.TestCase):
         dlml.CsvDataSet(
             file_pattern=['gs://file1.txt'],
             schema=schema),
-        cloud=True).wait()
+        cloud=True,
+        project_id='junk_project_id').wait()
       self.assertIn('Schema contains an unsupported type %s.' % col_type,
                     job.fatal_error.message)
 

